### ISSUE TITLE
Inkscape crash when using Numix Icon Theme

### ISSUE THREAD

**Issue by sagrawalx** [numixproject/numix-icon-theme#1160]
I was using the Numix Icon Theme on Xubuntu 16.04. Whenever I ran Inkscape (0.91) and hit "Layer > Layers...", Inkscape would repeatedly crash with the error message "Inkscape has encountered an internal error and will close now" (see screenshot). 

![inkscape-error](https://cloud.githubusercontent.com/assets/11269129/20040128/a90292ca-a406-11e6-86f4-5cf53f456991.png)

Running Inkscape from command line, I saw the following error message. 

```
terminate called after throwing an instance of 'Gtk::IconThemeError'

Emergency save activated!
Emergency save completed. Inkscape will close now.
If you can reproduce this crash, please file a bug at www.inkscape.org
with a detailed description of the steps leading to the crash, so we can fix it.
Aborted (core dumped)
```

This led me to change my icon theme, and then there were no errors. In fact, none of the other icon themes I have installed led to this error, so my guess is that this is an issue with Numix? 

In case it helps, gdb gives the following output. 

```
Starting program: /usr/bin/inkscape 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe5b17700 (LWP 18330)]
[New Thread 0x7fffe5316700 (LWP 18331)]
[New Thread 0x7fffe4b15700 (LWP 18332)]
[New Thread 0x7fffe3494700 (LWP 18333)]
[New Thread 0x7fffe2c93700 (LWP 18334)]
terminate called after throwing an instance of 'Gtk::IconThemeError'

Thread 1 "inkscape" received signal SIGABRT, Aborted.
0x00007fffefbf7428 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/unix/sysv/linux/raise.c:54
54	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00007fffefbf7428 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007fffefbf902a in __GI_abort () at abort.c:89
#2  0x00007ffff075684d in __gnu_cxx::__verbose_terminate_handler() ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff07546b6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff0754701 in std::terminate() ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff0754919 in __cxa_throw ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff7823e67 in Gtk::IconThemeError::throw_func(_GError*) ()
   from /usr/lib/x86_64-linux-gnu/libgtkmm-2.4.so.1
#7  0x00007ffff5758d12 in Glib::Error::throw_exception(_GError*) ()
   from /usr/lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#8  0x00007ffff782473d in Gtk::IconTheme::load_icon(Glib::ustring const&, int, Gtk::IconLookupFlags) const () from /usr/lib/x86_64-linux-gnu/libgtkmm-2.4.so.1
#9  0x0000000000a605ee in ?? ()
#10 0x000000000095d031 in ?? ()
#11 0x000000000095f1cb in ?? ()
#12 0x000000000089847d in ?? ()
#13 0x0000000000892e49 in ?? ()
#14 0x0000000000893a87 in ?? ()
#15 0x0000000000893c13 in ?? ()
```

**Comment by Foggalong** [comment]
@wa4557 Guessing this is something to do with the Inkscape icons we've added :/

**Comment by hlechner** [comment]
@Foggalong have you found a way to reproduce the issue?

No crash to me.

---

@sagrawalx could help me to reproduce this bug? It didn't crash to me.

I would like to ask you some questions, if you can please give me this info so I can dig into this bug.

- Have you tried with icon theme ``Numix`` (base iconset) or maybe ``Numix Circle`` or ``Numix Square``? if you have tried just with Circle or Square, could you please try the ``Numix`` only?
- Have you tried to run the Inkscape in a different DM? (if you use xfce for example, try a gnome if possible)
- Could you create a new user or just switch to a different one and execute the Inkscape to test it? maybe it is related to a specific configuration of your user.
- Could you provide where did you install the Numix icon theme? (if via git or repo for example) and also the version of installed one.

**Comment by Foggalong** [comment]
@sagrawalx I've not managed to reproduce this either :/

**Comment by ptizzy** [comment]
I am able to reproduce this on Arch, running Inkscape 0.92.  I was using Numix Circle, but switched to Numix to test.  I updated to the latest Numix icon theme from Git. It fails in Gnome and Openbox, under my normal user, and a fresh user account.

**Comment by bil-elmoussaoui** [comment]
@Foggalong I believe this must be reported upstream. Having those icons on Numix theme shouldn't harm in any way the way Inkscape works :( anyway, can anyone of you guys try using the dev version of inkscape https://inkscape.org/en/download/development-versions/? 
Anyway it seems to be already reported upstream https://bugs.launchpad.net/inkscape/+bug/1639605
https://bugs.launchpad.net/inkscape/+bug/1659193

**Comment by Foggalong** [comment]
I've posted a link to that issue here to see if we can have a conversation with the devs to try and source the actual problem. Will leave this issue open until we find out if it's a problem on our end or theirs :)

**Comment by hlechner** [comment]
Hello @ptizzy thanks for this information!

I'm also an Arch Linux user using these themes:
![image](https://cloud.githubusercontent.com/assets/11244067/22482166/efba18c0-e7de-11e6-9648-5cfe18c1c434.png)

I've tried the Numix icon theme from the repo and also the git version

I'm running it through Gnome 3.22 and using Inkscape 0.92, do you have any insight on how I can reproduce it? no error at all from here, and reproduce the issue is the key to solve it.

If you have no idea maybe some package from AUR? custom version of something that I'm using stock and that's why i can't reproduce it.

This following command can provide the packages installed locally (AUR packages will be listed there):
```
pacman -Qm
```
or
```
yaourt -Qm
```

**Comment by ptizzy** [comment]
@hlechner I wonder if this might be somehow related to the fact that I was running infinality.  I have moved over to official freetype2, but I'm wondering if there isn't some lingering configuration somewhere causing problems.

Adwaita and Gnome seem to be the only icon themes that work.  Using the Numix icon theme, Inkscape crashes if I try to open any dialog (Layers, Align & Distribute, Export to Bitmap, etc).  Using Elementary seems to be ok with most dialogs, but crashes when opening the Export to Bitmap dialog.  When I open Inkscape, using Numix or Elementary, several of the icons in the toolbar don't display correctly, and don't fallback to gnome, but display as a generic icon. I also noticed this with several icons in the Align & Distribute dialog.

I also tried Papirus icon theme with similar results.

Here is the output of pacman -Qm:
acroread 9.5.5-6
adobe-air 2.6-1
adobe-air-sdk 2.6-10
autorandr-git 20120301-1
bittorrent 5.2.2-4
canon-ufr 2.70-2
clutter-gst2 2.0.18-1
compton-git 0.1_beta2.83.g2343e4b-1
cower 16-1
cython2-kivy 0.23-1
ddrutility 2.8-1
dropbox 17.4.33-1
epson-inkjet-printer-201106w 1.0.1-8
exif 0.6.21-1
ezxml 0.8.6-1
faba-icon-theme-git 294.ab92a4c-2
faba-mono-icons-git 94.560b521-1
faience-themes 0.7-1
farstream-0.1 0.1.2-5
ffmpeg0.10 0.10.16-2
flattr-icon-theme-git 566.7d02906-1
fluidr3 122501-7
fritzing 0.9.3b-3
fslint 2.46-1
geoclue 0.12.99-3
gimp-plugin-bimp 1.17-1
gimp-plugin-layerfx 20120412-1
gimp-plugin-saveforweb 0.29.3-2
glue 0.9.4-1
goocanvas1 1.0.0-4
gpgmepp 16.08.3-1
grub-legacy 0.97-25
gstreamer0.10 0.10.36-4
gstreamer0.10-bad 0.10.23-20
gstreamer0.10-bad-plugins 0.10.23-20
gstreamer0.10-base 0.10.36-3
gstreamer0.10-base-plugins 0.10.36-3
gstreamer0.10-ffmpeg 0.10.13-2
gstreamer0.10-good 0.10.31-11
gstreamer0.10-good-plugins 0.10.31-11
gstreamer0.10-python 0.10.22-2
gstreamer0.10-ugly 0.10.19-16
gstreamer0.10-ugly-plugins 0.10.19-16
gtk-engine-unico 1.0.2-6
gtk-theme-numix-ocean 2.0.2-2
gtk-theme-numix-osx 2.ea4c2bd-1
gtkhtml4 4.10.0-1
hsqldb-java 1:1.8.0.10-2
ido 12.10.2-4
indicator-datetime 12.10.2-5
indicator-pantheon-session-bzr r241-1
indicator-power 12.10.5-3
indicator-sound 12.10.1-5
jmtpfs 0.5-1
jpnevulator 2.1.3-1
lastpass 4.1.23-1
lib32-gtk-engine-murrine 0.98.2-2
lib32-libusb-compat 0.1.5-3
lib32-llvm-amdgpu-lib-snapshot 20130403-2
lib32-pangox-compat 0.0.2-1
libbonobo 2.32.1-4
libfetch 2.33-3
libgee06 0.6.8-1
libmusicbrainz4 4.0.3-2
libsexy 0.1.11-5
libsignon-qt4 8.58-2
libzeitgeist 0.3.18-4
llvm-amdgpu-lib-snapshot 20130403-3
mash 0.2.0-3
musicbrainz 2.1.5-7
netkit-bsd-finger 0.17-9
numix-circle-icon-theme-git 0.r3523.cad3d39-1
numix-icon-theme-git 0.r1899.271471c79-1
numix-themes-archblue 2.5.1-1
numix-themes-git 2.6.0.r9.0d05abe-1
nwjs-bin 0.20.2-1
obshutdown 0.1_rc1-2
orbit2 2.14.19-3
pantheon-notify-bzr r27-1
papirus-icon-theme-git 1:r790.8f6a7eff3-1
pidgin-gnome-shell-extension-git 96.c10caca-1
plank-bzr r1578-1
pngcheck 2.3.0-2
pngnq 1.1-2
powerline-fonts-git r100.a44abd0-3
prelink 20130503-7
putty-freedesktop 3-1
pygoocanvas 0.14.1-7
pyorbit 2.24.0-5
python-mpd2 0.5.5-1
python-pyalsaaudio 0.8.2-1
python2-more_itertools 2.2-5
python2-pyalsaaudio 0.8.2-1
python2-pyo 0.8.2-1
python2-pypdf2 1.26.0-1
python2-pystache-git 1063.17a5dfd-1
pyxml 0.8.4-10
qt5-jsbackend 5.1.1-5
qtresistors 0.1.3-2
qtwebkit 2.3.4-4
ruby-maruku 0.7.2-3
ruby-sass 3.4.22-1
ruby-yard 0.8.7.6-1
samsungctl 0.6.0-2
shotcut-bin 170201-1
slock-xtra-git 20141117.191388c-1
sonata-git v1.7a2.r77.g4e34e07-1
stapler-git 0.3.2.r16.g7c153e6-1
teamviewer 12.0.71510-6
terminix 1.4.2-2
todotxt 2.10-3
ttf-ms-fonts 2.0-10
ttf-noto 20160917-1
ttf-raleway-font-family 3.0-1
uboot-mkimage 2014.07-1
udisks 1.0.5-3
vim-pathogen 2.4-1
volumeicon-pulsefix 20140305_1-1
wxgtk2.8 2.8.12.1-6
xchat-gnome-git 20120109-1
ympd-git 1.3.0.r0.gec008a4-1

**Comment by ptizzy** [comment]
After digging around, I found this post, which suggests infinality is causing the problem (by breaking harfbuzz):  https://bbs.archlinux.org/viewtopic.php?id=223031

I double-checked to make sure I had proper versions of fontconfig and freetype2.  Then, based on https://bugs.archlinux.org/task/52453, I ran /usr/bin/gdk-pixbuf-query-loaders --update-cache.  I don't know if that is what solved it or not, but it seems to be all fine now.
